nodewatcher/modules/frontend/ip_space/static/js/draw_cidrs.js

```diff
@@ -0,0 +1,298 @@
+                return url;
+            }
+        }
+        else {
+            if (typeof value !== 'undefined' && value !== null) {
+                var separator = url.indexOf('?') !== -1 ? '&' : '?';
+                hash = url.split('#');
+                url = hash[0] + separator + key + '=' + value;
+                if (typeof hash[1] !== 'undefined' && hash[1] !== null)
+                    url += '#' + hash[1];
+                return url;
+            }
+            else
+                return url;
+        }
+    }
+
+    draw(subnet, description) {
+        var times = 32 - parseInt(subnet.split('/')[1]);
+        var subnet_ips = Math.pow(2, times);
+        var shape_size = Math.sqrt(subnet_ips) / this.ips_pixel;
+
+        var start_num = this.ip2num(subnet.split('/')[0]) - this.offset;
+
+        var start_xy = d2xy(start_num);
+
+        var x = start_xy.x / this.ips_pixel;
+        var y = start_xy.y / this.ips_pixel;
+
+        this.svg.append('rect').attr('x', x).attr('y', y).attr('height', shape_size).attr('width', shape_size).style('fill', this.rgbToHex(times * 8, times * 8, times * 8)).style('opacity', 0.3).attr('id', subnet).attr('n', subnet_ips).attr('d', description);
+    }
+
+    loadPrefix(i, url) {
+        var self = this;
+        $.getJSON(url, function(data) {
+            if(data.next != null){
+                self.loadPrefix(i, data.next);
+                if(self.data[i] == undefined){
+                    self.data[i] = data.results;
+                }
+                else{
+                    self.data[i].concat(data.results);
+                }
+                for (var j = 0; j < data.results.length; j++) {
+                    self.draw(data.results[j].network + '/' + data.results[j].prefix_length, data.results[j].description);
+                }
+            }
+            else{
+                self.loadedPrefixes++;
+                if(self.data[i] == undefined){
+                    self.data[i] = data.results;
+                }else{
+                    self.data[i].concat(data.results);
+                }
+                for (var j = 0; j < data.results.length; j++) {
+                    self.draw(data.results[j].network + '/' + data.results[j].prefix_length, data.results[j].description);
+                }
+                if (self.loadedPrefixes == 32) {
+                    jQuery('rect').tipsy({
+                        gravity: 'w',
+                        html: true,
+                        title: function() {
+                            return this.id + '<br>Number of hosts: ' + $(this).attr('n') + '<br>Network name: ' + $(this).attr('d');
+                        }
+                    });
+                    self.setTopNodes(self.calcTopNodes(self.data));
+                }
+            }
+
+        });
+    }
+
+    load() {
+        for (var i = 0; i < 33; i++) {
+            this.loadPrefix(i, 'api/v2/pool/ip/?format=json&prefix_length=' + i);
```

Comment:
Get URL API from server-side and render-it into a template. See examples elsewhere how to do that.

Reply:
You do not have to add `format=json` because `getJSON` adds needed headers already.